$NetBSD: patch-src_nvim_CMakeLists.txt,v 1.1 2023/04/25 23:36:19 nikita Exp $

Upstream commit f5bf29a0ad01c53dcf67e22ef66981a1fd9a0939
build: include all dependency directories when generating headers

This will add all interface include directories property from all
targets to main_lib. This may not be universally wanted, in which case
we can revisit/rework it.

Closes https://github.com/neovim/neovim/issues/23237.

(cherry picked from commit 096e91b9888491bb845f9aadd3f64f3fe2dac394)

--- src/nvim/CMakeLists.txt
+++ src/nvim/CMakeLists.txt
@@ -387,11 +387,14 @@ foreach(gen_cdef DO_NOT_DEFINE_EMPTY_ATTRIBUTES ${prop})
   endif()
 endforeach()
 
-get_target_property(prop main_lib INTERFACE_INCLUDE_DIRECTORIES)
-foreach(gen_include ${prop})
-  list(APPEND gen_cflags "-I${gen_include}")
+get_directory_property(targets BUILDSYSTEM_TARGETS)
+foreach(target ${targets})
+  get_target_property(prop ${target} INTERFACE_INCLUDE_DIRECTORIES)
+  foreach(gen_include ${prop})
+    list(APPEND gen_cflags "-I${gen_include}")
+  endforeach()
 endforeach()
-list(APPEND gen_cflags "-I${DEPS_PREFIX}/include")
+
 if(APPLE AND CMAKE_OSX_SYSROOT)
   list(APPEND gen_cflags "-isysroot")
   list(APPEND gen_cflags "${CMAKE_OSX_SYSROOT}")
